packages/nimble-components/src/table/specs/table-column-specs/table-column-menu-button.md

```diff
@@ -0,0 +1,207 @@
+        -   slot's name: _unique-column-id_ + _column-requested-slot-name from column internal's slotNames array_ (e.g. table-column-slot2-menu)
+        -   slot's slot: _column-requested-slot-name from column internal's slotNames array_ (e.g. menu)
+    -   menu button column cell view - slot created within the menu button component
+        -   slot's name: _column-requested-slot-name from column internal's slotNames array_ (e.g. menu)
+        -   slot's slot: `"menu"`, which is the slot name required by the `nimble-menu-button` comonent
+-   Add events to pass slot information between cell views, rows, and the table:
+    -   `cell-view-slots-request` event
+        -   Fired by cell view instances if they want to request that the column's slots to be placed within that cell. For example, the menu button column's cell view will fire this event when the cell's `menu-button` fires a `beforetoggle` event with the `newState` as `true`.
+        -   Handled by table rows. When handling this event, a table row will fire the `row-slots-request` event to request the table to slot the necessary elements within the row.
+        -   Event details:
+            <!-- prettier-ignore -->
+            ```ts
+            interface CellViewSlotRequestEventDetail {
+                slots: { slot: string, name: string }[];
+            }
+            ```
+    -   `row-slots-request` event
+        -   Fired by table rows to request that column's slots to be placed within that row. This event is fired in response to the row recieving a `cell-view-slots-request` event fired by one of its cell views.
+        -   Handled by table. When handling this event, the table will update its template to move the appropriate slots into the correct table row.
+        -   Event details:
+            <!-- prettier-ignore -->
+            ```ts
+            interface RowSlotRequestEventDetail {
+                columnInternalId: string;
+                rowId: string;
+                slots: { slot: string, name: string }[];
+            }
+            ```
+
+## Alternative Implementations / Designs
+
+_Alternative #1_
+
+Rather than creating a menu button column, we could create a select column. However this has a number of challenges steming from the fact that the `nimble-select` has a different slotting model than the `nimble-menu-button`. As a result, it would be very difficult have an API that allows providing unique options for each record without a significant redesign of the select. Additionally, the `nimble-select` has interactions that allow the value to be changed without the component being open. This poses a challenge of handling all the cases where a given select's options need to be updated.
+
+## Open Issues
+
+-   In the Software Table, how should the currently installed version be represented? Currently it is italicized in both the menu and menu-button, but this is probably not the solution we want in the nimble table column.
```

Comment:
The design isn't finalized, but one good option will be to use the menu headers to segment the software between available and installed software.
<img width="792" alt="Screenshot 2024-05-30 at 4 33 42 PM" src="https://github.com/ni/nimble/assets/1458528/9f98b126-4553-4408-b26c-6dc5373e1765">

Reply:
I'll remove the open question since this design direction won't have any impact on the API for the column.